# CMML-diversity

This repository contains the code for the analysis of single-cell RNA sequencing data from CD34+ bone marrow mononuclear cells of patients with chronic myelomonocytic leukemia (CMML) described in:  

Ferrall-Fairbanks MC*, Dhawan A*, Johnson B, Newman H, Volpe V, Letson C, Ball M, Hunter A, Balasis M, Kruer T, Ben-Crentsil N, Kroeger JL, Balderas R, Komrokji RS, Sallman DA, Zhang J, Bejar R, Altrock PM, Padron E. "HSC depletion and expansion of inflammatory GMP cells are clinically relevant features of disease progression in chronic myelomonocytic leukemia (CMML)" (Submitted)

This work is also being presented at the 2021 American Society of Hematology (ASH) Annual Meeting on December 11 at 4pm. The full abstract for the oral presentation entitled "Single Cell Deconvolution of Myeloblasts Reveals Depletion of HSCs and Expansion of Inflammatory Granulocyte-Monocyte Progenitors (GMPs) Associated with Adverse Outcomes in Chronic Myelomonocytic Leukemia (CMML)" is available [here](https://ash.confex.com/ash/2021/webprogram/Paper151348.html). 

Code is written in R and utilizes Linux platforms for some applications. 

## Getting Started
The `src` folder contains subfolders with code for the primary and secondary analyses. 

![image](/_assets/CMML-scRNAseq-workflow.jpg)

## Contributing
If you would like to improve this code or have any feedback, please file an issue or pull request or you can contact the maintainer at mferrall.fairbanks@bme.ufl.edu. 
